File: hw/santorum/u3/hw32Strategy/Strategy/src/main/java/ec/edu/espe/strategy/model/InsertionSort.java
package ec.edu.espe.strategy.model;

import java.util.Arrays;

/**
 *
 * @author Thais Santórum
 */
public class InsertionSort implements SortingStrategy {
    
    public int[] sort(int data[]){
        System.out.println("Selected strategy InsertionSort");
        int[] arr = data.clone();
        Arrays.sort(arr);
        return arr; 
    }
    
}

File: exam/aymacana/u3/Q47_Strategy/model/insertion_sort.py
from model.sorting_strategy import SortingStrategy

class InsertionSort(SortingStrategy):

    def sort(self, data):
        for i in range(1, len(data)):
            key = data[i]
            j = i - 1
            while j >= 0 and data[j] > key:
                data[j + 1] = data[j]
                j -= 1
            data[j + 1] = key
        return data
